hazelcast/src/main/java/com/hazelcast/internal/metrics/MetricsUtil.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.metrics;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import javax.annotation.Nonnull;
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * A utility to escape and parse metric key in tagged format:<pre>
+ *     [tag1=value1,tag2=value2,...]
```

Comment:
The format is subject to change.

Escaping is not planned to be supported. Tag names and values as well as the metric name should not include `=` and space. The caller of a registration has to make sure this is the case. This limitation is most likely still to loose as we have to plan for smooth support when working with different storage technologies. Therefore recommendation is to keep this alphanumeric only. Such a util as this should not exist.

Reply:
The value can be a user input, for example a vertex name, map name. We don't limit those, we don't want to push limitations down to everything that uses metrics. It's much easier to work without limits, the `value` can store any string.

Reply:
@jbee what would be the problem if we would push escaping down to the metrics system? This way metrics can decide how to escape instead of letting every usage of the metrics deal with escaping.

Reply:
Its a complexity that should not be there. Only dynamic things can have user given tag values. You have to programmatically register those anyway so there is the perfect place to use all domain knowledge available and come up with the best value you can express given the limitations. Problem solved at the border. With escaping all components handling metrics would need to think about how to handle this special case. Problem taken to all places involved. Obviously I dislike this for such a unnecessary "feature".